src/services/expensesServices.js

```diff
@@ -0,0 +1,62 @@
+const { Expense } = require('./../models/Expense.model');
+const { Op } = require('sequelize');
+const normalize = ({ amount, category, id, note, spentAt, title, userId }) => {
+  return {
+    amount,
+    category,
+    id,
+    note,
+    spentAt,
+    title,
+    userId,
+  };
+};
+
+const getAll = async ({ categories, from, to, ...query }) => {
+  const where = { ...query };
+
+  if (categories) {
+    where.category = categories;
+  }
+
+  if (from && to) {
+    where.spentAt = { [Op.between]: [new Date(from), new Date(to)] };
+  } else if (from) {
+    where.spentAt = { [Op.gte]: [from] };
+  } else if (to) {
+    where.spentAt = { [Op.lte]: [to] };
+  }
+
+  return Expense.findAll({ raw: true, where });
+};
+
+const getById = (id) => {
+  return Expense.findByPk(id);
+};
+const create = (userId, spentAt, title, amount, category, note) => {
+  return Expense.create({
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  });
+};
+
+const remove = async (id) => {
+  await Expense.destroy({ where: { id } });
+};
+
+const update = (id, param = {}) => {
+  return Expense.update(param, { where: { id } });
```

Comment:
Consider adding error handling for the `update` function to manage potential exceptions, such as when the expense does not exist or the update fails.